README.md

```diff
@@ -0,0 +1,112 @@
+
+### Creating the cluster
+Once you've downloaded all the files of this repository ( `git clone https://github.com/fatz/dcos-secure-instance-profiles && cd dcos-secure-instance-profiles` ) you need to initialize terraform and start creating the cluster
+
+Before you start creating the cluster make sure your AWS setup is finished and working. Either `$AWS_PROFILE` needs to be set to the profile you want to use or make sure you've properly setup your aws cli `aws configure`. To ensure you will use the expected account you can run `aws sts get-caller-identity` and see the account id you will be using.
+
+```bash
+terraform init -upgrade .
+terraform apply
+```
+
+### If not already done download the dcos-cli
+
+```bash
+# on OSX
+brew install dcos-cli
+
+# on linux
+
+```
+
+### Attach to cluster
+After successfully creating the cluster we have to attach to the cluster
+
+```bash
+# in this setup we have to use --insecure as we did not give the load balancer a ACM cert and so it is an self signed one.
+dcos cluster setup $(terraform output masters_dns_name) --password=deleteme --username=bootstrapuser --insecure
+```
+
+### Ensure enterprise CLI
+Lets make sure we have the enterprise features available in our CLI ( this is usually just needed for older versions of DC/OS (cli))
+
+```bash
+dcos package install dcos-enterprise-cli --cli --yes
+```
+
+### AWS config secret
+We already prepared the aws config for the application in our `main.tf`. Next we create the secret from it.
+
+```bash
+dcos security secrets create /instance-profile-app/aws-config -v "$(terraform output secret_aws_conf)"
+```
+
+### Install marathon-lb
+To access our app lets install marathon-lb. As we’re running strict mode we’ve to create a service-account and a service-account-secret
+
+#### Prepare service account and secret
+
+```bash
+dcos security org service-accounts keypair mlb-private-key.pem mlb-public-key.pem
+dcos security org service-accounts create -p mlb-public-key.pem -d "Marathon-LB service account" marathon-lb-sa
+dcos security secrets create-sa-secret --strict mlb-private-key.pem marathon-lb-sa marathon-lb/service-account-secret
+dcos security org users grant marathon-lb-sa dcos:service:marathon:marathon:services:/ read
+dcos security org users grant marathon-lb-sa dcos:service:marathon:marathon:admin:events read --description "Allows access to Marathon events"
+```
+
+#### Install marathon-lb
+```bash
+echo '{"marathon-lb": {"secret_name": "marathon-lb/service-account-secret","marathon-uri": "https://marathon.mesos:8443"}}' | dcos package install marathon-lb --options=/dev/stdin --yes
+```
+
+### Deploy the marathon app
+The last step is to finally deploy our simple app using the bucket we prepared. We’re using the template given in our terraform file. You can simply review it using terraform output marathon_app_definition.
```

Comment:
The last step is to finally deploy our simple app using the bucket that we've prepared. We’re using the template given in our terraform file. You can review it simply by using the terraform output marathon_app_definition.